cpp/client/src/config.cpp

```diff
@@ -11,479 +11,414 @@
 
 #include "config.h"
 
+#define SECTION_GENERAL_KEY "general"
+#define CONSUMER_KEY "consume"
+#define SECTION_BASIC_KEY "basic"
+#define SECTION_AUDIO_KEY "audio"
+#define SECTION_VIDEO_KEY "video"
+#define SECTION_STREAMS_KEY "streams"
+#define CONSUMER_BAISC_STAT_KEY "stat"
+#define SECTION_STREAM_TYPE_AUDIO "audio"
+#define SECTION_STREAM_TYPE_VIDEO "video"
+
 using namespace std;
 using namespace libconfig;
 using namespace ndnrtc;
 using namespace ndnrtc::new_api;
 using namespace ndnlog;
 
-int loadParamsFromFile(const string &cfgFileName, ParamsStruct &params,
-                       ParamsStruct &audioParams)
-{
-    Config cfg;
+int lookupNumber(const Setting &SettingPath,string lookupKey, double &paramToFind);
+int lookupNumber(const Setting &SettingPath, string lookupKey, int &paramToFind);
+int lookupNumber(const Setting &SettingPath, string lookupKey, unsigned int &paramToFind);
+int loadConfigFile(const string &cfgFileName, 
+                    Config &cfg);
+int loadBasicConsumerSettings(const Setting &settingPath,
+                        GeneralConsumerParams& consumerGeneralParams);
+int loadBasicStatSettings(const Setting &consumerBasicStatSettings, 
+                        std::vector<Statistics> &statistics);
+int loadGeneralSettings(const Setting &general, 
+                        GeneralParams &generalParams);
+int loadMediaStreamParams(const Setting &consumerStreamSettings, 
+                            MediaStreamParamsSupplement* defaultAudioStream);
+int loadAudioThreadParams(const Setting &consumerStreamThreadsSettings, 
+                            AudioThreadParams* audioThread, 
+                            MediaStreamParamsSupplement* defaultAudioStream);
+int loadVideoThreadParams(const Setting &consumerStreamThreadsSettings, 
+                            VideoThreadParams* videoThread, 
+                            MediaStreamParamsSupplement* defaultVideoStream);
+
+
+int loadParamsFromFile(const string &cfgFileName,
+                       ClientParams &params){
     
-    // Read the file. If there is an error, report it and exit.
-    try
-    {
-        cfg.readFile(cfgFileName.c_str());
-    }
-    catch(const FileIOException &fioex)
-    {
-        LogError("")
-        << "I/O error while reading file." << endl;
-        return(EXIT_FAILURE);
-    }
-    catch(const ParseException &pex)
-    {
-        LogError("")
-        << "Parse error at " << pex.getFile()
-        << ":" << pex.getLine()
-        << " - " << pex.getError() << endl;
-        
-        return(EXIT_FAILURE);
+    Config cfg;
+
+    if (loadConfigFile(cfgFileName, cfg)==EXIT_FAILURE) {
```

Comment:
if your `loadConfigFile` can throw exceptions, it should be wrapped in `try...catch` block here. see http://www.stroustrup.com/bs_faq2.html#exceptions-i